src/SAWScript/Yosys/Utils.hs

```diff
@@ -259,3 +262,39 @@ eqBvRecords sc cty a b = do
       , "\nhas no fields"
       ]
     (e:es) -> foldM (\x y -> liftIO $ SC.scAnd sc x y) e es
+
+-- | Encode the given string such that is a valid Cryptol identifier. 
+-- Since Yosys cell names often look like "\42", this makes it much easier to manipulate state records,
+-- which are keyed by cell name.
+cellIdentifier :: Text -> Text
+cellIdentifier = Text.pack . zEncodeString . Text.unpack
+
+-- | Build a SAWCore type corresponding to the Cryptol record type with the given field types
+fieldsToType ::
+  MonadIO m =>
+  SC.SharedContext ->
+  Map Text (SC.Term, C.Type) ->
+  m SC.Term
+fieldsToType sc = cryptolRecordType sc . fmap fst
+
+-- | Build a Cryptol record type with the given field types
+fieldsToCryptolType ::
+  MonadIO m =>
+  Map Text (SC.Term, C.Type) ->
+  m C.Type
+fieldsToCryptolType fields = pure . C.tRec . C.recordFromFields $ bimap C.mkIdent snd <$> Map.assocs fields
+
+-- | Given a bit pattern ([Bitrep]) and a term, construct a map associating that output pattern with
```

Comment:
Should "`[Bitrep]`" actually be "`[b]`" here? I don't see any occurrences of `Bitrep` in the type signature.